infrastructure/tari_util/src/cipher.rs

```diff
@@ -20,59 +20,32 @@
 // WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
 // USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-use crate::chacha20::*;
-
-pub trait Cipher {
-    /// Encode using a cipher keystream with a default nonce
-    fn encode(&self, key: &[u8; 32]) -> Vec<u8>;
-
-    /// Encode using a cipher keystream with a self selected nonce
-    fn encode_with_nonce(&self, key: &[u8; 32], nonce: &[u32; 3]) -> Vec<u8>;
-
-    /// Decode using a cipher keystream with a default nonce
-    fn decode(bytes: &[u8], key: &[u8; 32]) -> Self
-    where Self: Sized;
-
-    /// Decode using a cipher keystream with a self selected nonce
-    fn decode_with_nonce(bytes: &[u8], key: &[u8; 32], nonce: &[u32; 3]) -> Self
-    where Self: Sized;
+use crate::{ByteArray, ByteArrayError};
```

Comment:
I think it will be appropriate for the "cipher.rs" and "chacha20.rs" files to live in a "cipher" subfolder in tari_util.

Reply:
Sure, makes sense if we add more interfaces.